src/bazel/bazel_utils.ts

```diff
@@ -144,3 +138,43 @@ export function getBazelWorkspaceFolder(fsPath: string): string | undefined {
 
   return undefined;
 }
+
+/**
+ * Search for the path to the directory that has the Bazel WORKSPACE file for
+ * the given file.
+ *
+ * If multiple directories along the path to the file have files called
+ * "WORKSPACE", the lowest path is returned.
+ *
+ * @param fsPath The path to a file in a Bazel workspace.
+ * @returns The path to the directory with the Bazel WORKSPACE file if found,
+ * others undefined.
+ */
+export function getBazelWorkspaceFolder(fsPath: string): string | undefined {
+  // These are the names of the files that mark the root of a repository
+  // or workspace.
+  const REPO_ROOT_FILE_NAMES = [
+    "MODULE.bazel",
+    "REPO.bazel",
+    "WORKSPACE.bazel",
+    "WORKSPACE",
+  ];
+
+  return getTargetFolderForFile(fsPath, REPO_ROOT_FILE_NAMES);
+}
+
+/**
+ * Search for the path to the directory that has the BUILD.Bazel file for
+ * the given file.
+ *
+ * @param fsPath The path to a file in a Bazel workspace.
+ * @returns The Bazel path to the directory with the BUILD.Bazel if found,
+ * others undefined.
+ */
+export function getBazelPackageFolder(fsPath: string): string | undefined {
+  // These are the names of the files that mark the root of a repository
+  // or workspace.
+  const PACKAGE_FILE_NAMES = ["BUILD.bazel"];
```

Comment:
Both variants are valid:
```suggestion
  const PACKAGE_FILE_NAMES = ["BUILD.bazel", "BUILD"];
```